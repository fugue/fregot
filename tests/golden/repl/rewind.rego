package rewind

crashing_rule {
  x = 1
  x + "hi" > "wat"
}

failing_rule {
  x = 1
  x = 2
}
