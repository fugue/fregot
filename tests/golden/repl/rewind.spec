{
    "command": "fregot",
    "arguments": ["repl", "--no-history-file", "rewind.rego"],
    "asserts": [
        {"exit_code": 0},
        {"stderr": "${SPEC_NAME}.stderr"},
        {"stdout": "${SPEC_NAME}.stdout"}
    ],
    "stdin": [
        ":break rewind.crashing_rule",
        "data.rewind.crashing_rule",
        ":next",
        ":next",
        ":rewind",
        ":quit",
        ":break rewind.failing_rule",
        "data.rewind.failing_rule",
        ":next",
        ":next",
        ":rewind",
        ":quit"
    ]
}
