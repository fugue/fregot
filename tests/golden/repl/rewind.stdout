repl% repl% 4|   x = 1
     [32m^^^^^[0m
rewind(debug)% 5|   x + "hi" > "wat"
     [32m^^^^^^^^^^^^^^^^[0m
rewind(debug)% [32m(debug)[0m error
rewind(error)% 5|   x + "hi" > "wat"
     [32m^^^^^^^^^^^^^^^^[0m
rewind(debug)% repl% repl% 9|   x = 1
     [32m^^^^^[0m
rewind(debug)% 10|   x = 2
      [32m^^^^^[0m
rewind(debug)% [32m(debug)[0m finished
repl% 10|   x = 2
      [32m^^^^^[0m
rewind(debug)% repl% 